### ISSUE TITLE
False positive: Tuple is unsuscriptable since pylint 2.9.3

### ISSUE THREAD

**Issue by LefterisJP** [PyCQA/pylint#4739]
### Steps to reproduce

Get the `web3==5.21.0` package. Then use this simple script.

```python
from web3 import Web3
from web3._utils.abi import get_abi_output_types

WEB3 = Web3()

address = '0x0'
abi = {'foo': 'boo'}
method_name = 'foo'
arguments = None
contract = WEB3.eth.contract(address=address, abi=abi)
fn_abi = contract._find_matching_fn_abi(
    fn_identifier=method_name,
    args=arguments if arguments else [],
)
output_types = get_abi_output_types(fn_abi)
result = b'dasdad'
decoded_method = WEB3.codec.decode_abi(output_types, result)

if len(decoded_method) == 1:
    a = decoded_method[0]
else:
    a = decoded_method

print(a)
```

### Current behavior

```
test5.py:20:8: E1136: Value 'decoded_method' is unsubscriptable (unsubscriptable-object)
```

### Expected behavior

No error. A tuple is returned and its length is checked.

### pylint --version output

pylint 2.9.5. First appeared in 2.9.3 as per [this](https://github.com/PyCQA/pylint/issues/4694#issuecomment-877593945) comment.

**Comment by DanielNoord** [comment]
I checked this for a little bit and it seems astroid thinks `decoded_method` is a generator. With my current (limited) knowledge I found no way to tell what the result of said generator was and if it would always be a tuple.

This might be something that needs to be fixed in `astroid` instead of `pylint` but I'm not sure. `pylint` 2.9.3 got released around the time `astroid` 2.6.2 was released, although those release notes don't have anything in them that stands out as a possible source of the problem to me.

**Comment by Pierre-Sassoulas** [comment]
By the look of it it look like we could git bisect the problem in astroid to pinpoint the commit that caused this.

**Comment by clavedeluna** [comment]
This is still reproduceable

```
pip install web3==5.21.0
... Value 'decoded_method' is unsubscriptable (unsubscriptable-object)
```

**Comment by LefterisJP** [comment]
yeah we still have this issue, but forgot about it since we just add pylint: ignore for it. If you guys figure it out at some point it would be really appreciated.
